### ISSUE TITLE
Question: How to type into selection required 'Edit' control types?

### ISSUE THREAD

**Issue by Avnsx** [pywinauto/pywinauto#1062]
I've a chatbox that I'm trying to write stuff into. It's control type is Edit. 
```
Edit - 'Type here...'    (L0, T0, R0, B0)
   |    |    |    | ['Edit2']
   |    |    |    | child_window(title="Type here...", control_type="Edit")
```
the title will always change, so I couldn't use that and instead did it like this:

```python
chatbox = app.child_window(control_type='Edit', found_index=1)
chatbox.set_edit_text(input) #inputs my text
chatbox.click_input() #help here
chatbox.type_keys('{ENTER}', pause=None, with_spaces=True, set_foreground=False) #because i am forced to use enter to send the message since the chatbox does not have a send message button
```
I am trying to send text through it; but it forces me to click on it to turn it "enabled". That statement is also not correct because if checking it's properties or even setting them manually which I believe works like this:
```
python
chatbox.has_keyboard_focus=True
chatbox.is_enabled=True
````
does not help me solve the problem either. I also ``tried click()`` and ``invoke()``, which didn't work either. The only thing that did work for me and "placed my cursor inside it" by clicking on the chatbox  was ``click_input()``, but I really don't like the animation it does since it moves the mouse to it to click it. Is there not a simpler way where I don't get a animation but am still able to have my cursor inside the chatbox to be able to "send" my text that i placed with ``set_edit_text()`` inside it before?

Also here some additional info about the chatbox:
```python
{'class_name': '', 'friendly_class_name': 'Edit', 'texts': [''], 'control_id': None, 'rectangle': <RECT L404, T928, R777, B969>, 'is_visible': True, 'is_enabled': True, 'control_count': 0, 'is_keyboard_focusable': True, 'has_keyboard_focus': True, 'automation_id': '', 'selection_indices': (0, 0)}
['WAIT_CRITERIA_MAP', '_WindowSpecification__check_all_conditions', '_WindowSpecification__get_ctrl', '_WindowSpecification__parse_wait_args', '_WindowSpecification__resolve_control', '__call__', '__class__', '__delattr__', '__dict__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__getitem__', '__gt__', '__hash__', '__init__', '__init_subclass__', '__le__', '__lt__', '__module__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', '__weakref__', '_ctrl_identifiers', 'actions', 'allow_magic_lookup', 'app', 'backend', 'child_window', 'criteria', 'dump_tree', 'exists', 'print_control_identifiers', 'print_ctrl_ids', 'wait', 'wait_not', 'window', 'wrapper_object']
```

**Comment by vasily-v-ryabov** [comment]
`has_keyboard_focus()` and `is_enabled()` are read-only methods, not properties. In theory method `set_focus()` should work, but method `type_keys()` already calls `set_focus()` inside by default (you can skip it by `type_keys()` param `set_foreground=False`).

For "win32" backend we have special method `set_keyboard_focus()` for such edit controls. But this may be not suitable for "uia" backend because edit elements may be not visible to "win32" backend at all.

Is it some publicly available chat application?

**Comment by Avnsx** [comment]
> `has_keyboard_focus()` and `is_enabled()` are read-only methods, not properties. In theory method `set_focus()` should work, but method `type_keys()` already calls `set_focus()` inside by default (you can skip it by `type_keys()` param `set_foreground=False`).
> 
> For "win32" backend we have special method `set_keyboard_focus()` for such edit controls. But this may be not suitable for "uia" backend because edit elements may be not visible to "win32" backend at all.
> 
> Is it some publicly available chat application?

Yes it's a game client lobby chatbox. More specific it's the chatbox that appears in the champion selection part of [league of legends](https://na.leagueoflegends.com/) a MOBA genre game.

Here's a picture of what it looks like(it's the small little box on the lower bottom side):
![League Client](https://i.imgur.com/t0ZHBT9.png)

**Comment by vasily-v-ryabov** [comment]
Try `chatbox.iface_value.SetValue(input)` instead of `chatbox.set_edit_text(input)`. It should set keyboard focus as I've figured out for other application. So you will not need `click_input()` then.

But I see in the code that `set_edit_text()` calls `SetValue()` inside. So it may not work for you.

BTW, why are you trying to eliminate `.click_input()`? Usually if you don't need `.click_input()`, you don't need `.type_keys()` as well.

**Comment by Avnsx** [comment]
> Try `chatbox.iface_value.SetValue(input)` instead of `chatbox.set_edit_text(input)`. It should set keyboard focus as I've figured out for other application. So you will not need `click_input()` then.
> 
> But I see in the code that `set_edit_text()` calls `SetValue()` inside. So it may not work for you.
> 
> BTW, why are you trying to eliminate `.click_input()`? Usually if you don't need `.click_input()`, you don't need `.type_keys()` as well.

Yes, ``chatbox.iface_value.SetValue(input)`` indeed doesn't work for me and I'm still having to use ``click_input()``🤔

I'm trying to eliminate ``click_input()`` because of the foced animation it does on which it moves the users mouse to the element. Can I somehow just disable the animation, which drags my mouse to the object ?
